torchaudio/functional.py

```diff
@@ -311,17 +312,21 @@ def create_fb_matrix(n_freqs, f_min, f_max, n_mels):
         ``A * create_fb_matrix(A.size(-1), ...)``.
     """
     # freq bins
-    freqs = torch.linspace(f_min, f_max, n_freqs)
+    # Equivalent filterbank construction by Librosa
+    all_freqs = torch.linspace(0.0, float(sample_rate) / 2., n_freqs)
```

Comment:
This construction generates `n_freqs` bins between 0 and sample_rate/2. Since the line after filters out what's outside of `f_min` and `f_max`, we may end up with less than `n_freqs` bins, right? Is that expected?

Also, what happens if there are no bins between `f_min` and `f_max` in the range? Do we get an error?

Reply:
It is expected.  If we do a STFT with N = 2048, we should expect 1025 frequency bins corresponding to the entire frequency range from DC (0) to Nyquist.  We should not have 1025 bins between an arbitrary f_min and f_max set by the user.

Our code for slopes / zero / down_slopes / up_slopes is quite complicated but it will still create a size [n_freqs, n_mels] data structure that we have verified against librosa implementation in the test.

Yes in general, f_min and f_max can be set to bad values depending on what n_mels, n_freqs are and sample rate are.  Even if f_max > f_min but only slightly greater, we could return some results.  In this case we will still produce a fb matrix but some filters will be entirely zero.  This is consistent with librosa behavior and I have added a test for it. 

However, librosa will also output the following warning message ( n_mel is 56, fmin is 800, fmax is 900, n_fft is 1025 and sample_rate is 22050.)

```
Empty filters detected in mel frequency basis. Some channels will produce empty responses. Try increasing your sampling rate (and fmax) or reducing n_mels.
  warnings.warn('Empty filters detected in mel frequency basis. '
```